right_arm:
  planner_configs:
    arastar_bfs_manip:
      search_config: ARAConfig
      heuristic_config: BFSConfig
      graph_config: RightArmManipLatticeConfig
      shortcut_config: JointSpaceShortcutConfig
    larastar_bfs_manip:
      search_config: LARAConfig
      heuristic_config: BFSConfig
      graph_config: RightArmManipLatticeConfig
      shortcut_config: JointSpaceShortcutConfig
    mhastar_bfs_manip:
      search_config: ARAConfig
      heuristic_config: BFSConfig
      graph_config: RightArmManipLatticeConfig
      shortcut_config: JointSpaceShortcutConfig
    arastar_bfs_workspace:
      search_config: ARAConfig
      heuristic_config: BFSConfig
      graph_config: RightArmWorkspaceLatticeConfig
      shortcut_config: JointSpaceShortcutConfig
    arastar_euclid_workspace:
      search_config: ARAConfig
      heuristic_config: EuclidDistConfig
      graph_config: RightArmWorkspaceLatticeConfig
      shortcut_config: JointSpaceShortcutConfig
    larastar_euclid_workspace:
      search_config: LARAConfig
      heuristic_config: EuclidDistConfig
      graph_config: RightArmWorkspaceLatticeConfig
      shortcut_config: JointSpaceShortcutConfig
    adaptive_planner:
      search_config: AdaptivePlannerConfig
      heuristic_config: BFSConfig
      graph_config: RightArmAdaptiveWorkspaceLatticeConfig
      shortcut_config: JointSpaceShortcutConfig

left_arm:
  planner_configs:
    arastar_bfs_manip:
      search_config: ARAConfig
      heuristic_config: BFSConfig
      graph_config: LeftArmManipLatticeConfig
      shortcut_config: JointSpaceShortcutConfig
    larastar_bfs_manip:
      search_config: ARAConfig
      heuristic_config: BFSConfig
      graph_config: LeftArmManipLatticeConfig
      shortcut_config: JointSpaceShortcutConfig
    mhastar_bfs_manip:
      search_config: ARAConfig
      heuristic_config: BFSConfig
      graph_config: LeftArmManipLatticeConfig
      shortcut_config: JointSpaceShortcutConfig

search_configs:
  ARAConfig:
    search: arastar
    epsilon: 100.0
    repair_time: 1.0
  LARAConfig:
    search: larastar
    epsilon: 100.0
  MHAConfig:
    search: mhastar
    epsilon: 10.0
    mha_epsilon: 10.0
  AdaptivePlannerConfig:
    search: padastar
    epsilon_plan: 100.0
    epsilon_track: 100.0

heuristic_configs:
  BFSConfig:
    heuristic: bfs
    bfs_res_x: 0.015
    bfs_res_y: 0.015
    bfs_res_z: 0.015
    bfs_sphere_radius: 0.02
  EuclidDistConfig:
    heuristic: euclid
  JointDistConfig:
    heuristic: joint_distance

# walker_right_arm.yaml
graph_configs:
  RightArmManipLatticeConfig:
    graph: manip
    discretization:
      right_limb_j1:    0.017453292519943295
      right_limb_j2:    0.017453292519943295
      right_limb_j3:    0.017453292519943295
      right_limb_j4:    0.017453292519943295
      right_limb_j5:    0.017453292519943295
      right_limb_j6:    0.017453292519943295
      right_limb_j7:    0.017453292519943295
    use_xyz_snap_mprim: false
    use_rpy_snap_mprim: true
    use_xyzrpy_snap_mprim: true
    use_short_dist_mprims: true
    xyz_snap_dist_thresh: 0.0 
    rpy_snap_dist_thresh: 0.1
    xyzrpy_snap_dist_thresh: 0.2
    short_dist_mprims_thresh: 0.4

  # left_arm
  LeftArmManipLatticeConfig:
    graph: manip
    discretization:
      left_limb_j1:    0.017453292519943295
      left_limb_j2:    0.017453292519943295
      left_limb_j3:    0.017453292519943295
      left_limb_j4:    0.017453292519943295
      left_limb_j5:    0.017453292519943295
      left_limb_j6:    0.017453292519943295
      left_limb_j7:    0.017453292519943295
    use_xyz_snap_mprim: false
    use_rpy_snap_mprim: true
    use_xyzrpy_snap_mprim: true
    use_short_dist_mprims: true
    xyz_snap_dist_thresh: 0.0
    rpy_snap_dist_thresh: 0.1
    xyzrpy_snap_dist_thresh: 0.2
    short_dist_mprims_thresh: 0.4

shortcut_configs:
  JointSpaceShortcutConfig:
    shortcutter: joint_space
    shortcut_path: true
    interpolate_path: true
